vignettes/na-notebook.Rmd

```diff
@@ -0,0 +1,510 @@
+---
+title: "na-notebook"
+output:
+  pdf_document: default
+  html_document: default
+date: "2024-08-19"
+---
+
+```{r setup, warning=FALSE, echo=FALSE}
+suppressPackageStartupMessages({
+  library(epidatr)
+  library(epiprocess)
+  library(dplyr)
+  library(tidyr)
+  library(ggplot2)
+})
+utils::globalVariables(c("is_na", "signal", "issue"))
+```
+
+# How Different NAs Arise and Strategies for Handling Them
+
+### NAs from merging
+
+First let's start with discussing the most common type of missing values that appeared in the context of my auxiliary signal project. When working with multiple signals each signal will likely begin recording at different times. In other words each signal's first data point $t_0$ will differ on the absolute time scale. As a result, when calling `epix_merge()` to combine multiple signals, the signals that started recording at a later point in time will have missing values for the time periods where the other signals were already recording. Here's a quick example
```

Comment:
todo: let's present this as more instructional.  Here, that means just tweaking language, getting rid of "that appeared in the context of my auxiliary signal project", and then softening/rewording "the most common" since it may not be the most common in general.

(But in other parts we actually need to make it the content more instructional/relevant (use real data not buggy/artificial).)

Reply:
issue: `epix_merge()` can introduce NAs both from differing min `time_value`s and differing min `version`s, but
- the description here is a little ambiguous about which it's referring to
- the table is showing something more like an `epi_df`

Possible fix: instead of mentioning `epix_merge()` at this point, we could say this is an issue with some types of joins, and then transition into the `epix_merge()` example with some more discussion.